### PR TITLE
Feture/tao 7389/execution context interface

### PR DESCRIPTION
**Pull request by SashaTAO**
Created interface to work with delivery execution context data

### CODE REVIEW

File: model/execution/DeliveryExecutionContext.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoDelivery\model\execution;
+
```
Comment (peetya) [inline model/execution/DeliveryExecutionContext.php]: Please remove empty line.

File: model/execution/DeliveryExecutionContext.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    private $label;
+
+    /**
+     * DeliveryExecutionContext constructor.
+     *
+     * @param string $executionId
+     * @param string $type
+     * @param string $label
+     */
+    public function __construct($executionId, $executionContextId, $type, $label)
```
Comment (peetya) [inline model/execution/DeliveryExecutionContext.php]: `$executionContextId` is missing from PHPDoc.

File: test/unit/model/execution/DeliveryExecutionContextTest.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoDelivery\test\unit\model\execution;
+
+use oat\generis\test\TestCase;
+use oat\taoDelivery\model\execution\DeliveryExecutionContext;
+use oat\taoDelivery\model\execution\DeliveryExecutionContextInterface;
+
+class DeliveryExecutionContextTest extends TestCase
```
Comment (peetya) [inline test/unit/model/execution/DeliveryExecutionContextTest.php]: If the setters will be different depending on the implementations, maybe it worth to cover them with unit tests. What do you think?

File: model/execution/DeliveryExecutionContext.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    {
+        $this->executionId = $executionId;
+        $this->executionContextId = $executionContextId;
```
Comment (peetya) [inline model/execution/DeliveryExecutionContext.php]: What do you think about validating these two parameters?

File: test/unit/model/execution/DeliveryExecutionContextTest.php (hunk 3 above)
Comment (SashaTAO) [reply test/unit/model/execution/DeliveryExecutionContextTest.php]: I created tests only methods with logic. Other methods for now are just simple getters/setters

File: model/execution/DeliveryExecutionContext.php (hunk 4 above)
Comment (SashaTAO) [reply model/execution/DeliveryExecutionContext.php]: Done

File: test/unit/model/execution/DeliveryExecutionContextTest.php [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+     */
+    public function testConstructThrowsException($executionId, $executionContextId, $type, $label)
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        new DeliveryExecutionContext($executionId, $executionContextId, $type, $label);
+    }
+
+    /**
+     * @param array $executionId
```
Comment (peetya) [inline test/unit/model/execution/DeliveryExecutionContextTest.php]: Wrong PHPDoc

**Pull request merged**